CondTools/BeamSpot/plugins/BeamSpotOnlineRecordsWriter.cc

```diff
@@ -0,0 +1,235 @@
+// -*- C++ -*-
+//
+// Package:    CondTools/BeamSpot
+// Class:      BeamSpotOnlineRecordsWriter
+//
+/**\class BeamSpotOnlineRecordsWriter BeamSpotOnlineRecordsWriter.cc CondTools/BeamSpot/plugins/BeamSpotOnlineRecordsWriter.cc
+
+ Description: EDAnalyzer to create a BeamSpotOnlineHLTObjectsRcd or BeamSpotOnlineLegacyObjectsRcd payload from a txt file and dump it in a db file
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Francesco Brivio
+//         Created:  Tue, 11 Feb 2020 11:10:12 GMT
+//
+//
+
+// system include files
+#include <memory>
+#include <string>
+#include <fstream>
+#include <iostream>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+
+#include "CondFormats/DataRecord/interface/BeamSpotOnlineLegacyObjectsRcd.h"
+#include "CondFormats/DataRecord/interface/BeamSpotOnlineHLTObjectsRcd.h"
+#include "CondFormats/BeamSpotObjects/interface/BeamSpotOnlineObjects.h"
+#include "CondCore/DBOutputService/interface/PoolDBOutputService.h"
+
+//
+// class declaration
+//
+
+class BeamSpotOnlineRecordsWriter : public edm::one::EDAnalyzer<> {
+public:
+  explicit BeamSpotOnlineRecordsWriter(const edm::ParameterSet&);
+  ~BeamSpotOnlineRecordsWriter() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  cond::Time_t pack(uint32_t, uint32_t);
+
+private:
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+  void endJob() override;
+
+  // ----------member data ---------------------------
+  bool fIsHLT;
+  std::ifstream fasciiFile;
+  std::string fasciiFileName;
+  uint32_t fIOVStartRun;
+  uint32_t fIOVStartLumi;
+  cond::Time_t fnewSince;
+  bool fuseNewSince;
+};
+
+//
+// constructors and destructor
+//
+BeamSpotOnlineRecordsWriter::BeamSpotOnlineRecordsWriter(const edm::ParameterSet& iConfig) {
+  //now do what ever initialization is needed
+  fIsHLT = iConfig.getParameter<bool>("isHLT");
+  fasciiFileName = iConfig.getUntrackedParameter<std::string>("InputFileName");
+  fasciiFile.open(fasciiFileName.c_str());
+  if (iConfig.exists("IOVStartRun") && iConfig.exists("IOVStartLumi")) {
```

Comment:
"IOVStartRun" and "InputFileName" parameters are defined in the `fillDescriptions` method, and therefore they will always exist.
As such, `fuseNewSince` can never be false: either the code can be simplified, or there is a flaw in the logic.